src/lib.rs

```diff
@@ -2,25 +2,87 @@
 
 extern crate alloc;
 
+use ark_bn254::{Bn254, Fq, Fr, G1Affine, G1Projective, G2Affine, G2Projective};
+use ark_ec::pairing::Pairing;
 use ark_ec::{pairing::PairingOutput, AffineRepr, CurveGroup};
+use ark_ff::{BigInteger256, Field, PrimeField, Zero};
+use ark_serialize::CanonicalSerialize;
+use ark_std::One;
+use consts::{BLS_SIG_KEYGEN_SALT, BN254_CURVE_ORDER};
 use errors::BLSError;
+use hkdf::Hkdf;
 use num_bigint::BigUint;
-use ark_std::One;
-use ark_bn254::{Bn254, Fq, Fr, G1Affine, G1Projective, G2Affine, G2Projective};
-use ark_ec::pairing::Pairing;
-use ark_ff::{BigInteger256, Field, PrimeField};
 use sha2::{Digest, Sha256};
 
+pub mod consts;
 pub mod errors;
+pub mod keystores;
+pub mod mnemonics;
+pub mod utils;
 
 fn pairing(u: G2Affine, v: G1Affine) -> PairingOutput<Bn254> {
     Bn254::pairing(v, u)
 }
 
+fn xmd_hash_function(input: &[u8]) -> Vec<u8> {
```

Comment:
can you add comments to these methods to explain what they do?

Reply:
added. I think it might be good to leave the comments to a different PR as there will be some more changes coming in.